Bottom: 8545f343f0092b3cead9c74ae8383f0e28784e55
Top:    2f5b9b6ed6a8de234fcae1bedfe828a856c38e4a
Author: Ian Kent <ikent@redhat.com>
Date:   2013-04-08 15:05:16 +0800

Refresh of autofs-5.0.7-fix-submount-tree-not-all-expiring.patch

---

diff --git a/lib/master.c b/lib/master.c
index 59356be..d769342 100644
--- a/lib/master.c
+++ b/lib/master.c
@@ -938,7 +938,6 @@ int master_notify_submount(struct autofs_point *ap, const char *path, enum state
 		if (this->state == ST_SHUTDOWN) {
 			this = NULL;
 			st_mutex_unlock();
-			mounts_mutex_unlock(ap);
 			break;
 		}
 
@@ -977,10 +976,11 @@ int master_notify_submount(struct autofs_point *ap, const char *path, enum state
 			st_mutex_lock();
 		}
 		st_mutex_unlock();
-		mounts_mutex_unlock(ap);
 		break;
 	}
 
+	mounts_mutex_unlock(ap);
+
 	return ret;
 }
